### ISSUE TITLE
Exporting a chart with exporting.getImage("png") loads all css of the site

### ISSUE THREAD

**Issue by calvaradocl** [amcharts/amcharts4#3356]
**Describe what you are trying to achieve and what you are having trouble with.**

We are using amCharts4 to generate some charts. When trying out the export function to create a PDF as the documentation suggests on https://www.amcharts.com/docs/v4/tutorials/generating-multi-content-pdf-export/ we are experiencing a strange issue where when the Promises are run, the site starts to load all the CSS stylesheets and fonts before returning the RAW image.

Our code is dynamic, and being tested on 4 charts that are on the page:

```
var promises = [];
$.each(this.widgetMaster, function (i, v) {
    if (v.props.widget.type.indexOf("Chart") != -1) {
        var some = v.state.chart.exporting.getImage("png");
        promises.push(some);
    }
});


Promise.all(promises).then(function (res) {               

    
    // Create document template
    var doc = {
        pageSize: "A4",
        pageOrientation: "portrait",
        pageMargins: [30, 30, 30, 30],
        content: []
    };

    for (var i = 1; i < res.length; i++) {
        doc.content.push({
            image: res[i]
        });
    }
                  
    pdfMake.createPdf(doc).download("report.pdf");
});

```

We tested and discovered the same behavior if adding additional CSS files on the default demo that the website offers, with the following results:

![image](https://user-images.githubusercontent.com/17859695/110227653-7623e900-7ed9-11eb-87a6-0111c3a904f8.png)

You can find the pen here:
https://codepen.io/calvaradocl/pen/BaQOxMr

It's a perfect clone of https://codepen.io/team/amcharts/pen/qvRMQj, with only 3 CSS fonts added. If you run the export and debug the console, this issue occurs.

In our environment, we have many fonts, and this occurs in the same way for every chart that is being exported as an image raw. Is there any workaround for this issue?

**Comment by martynasma** [comment]
Well, in order to be able to export chart with correct fonts they need to be loaded. You can disable it using `useWebFonts` setting.

https://www.amcharts.com/docs/v4/concepts/exporting/#Web_fonts

**Comment by calvaradocl** [comment]
Hi @martynasma,

Thanks for the guidance. I understand that in the end, the AmCharts engine wants to ensure that if certain fonts are used, they should be load correctly. 

Maybe an option to set which fonts are being used for customizing charts, as an index, could be better. As an example, in the attached pen, those 3 CSS font icons are never used on the chart. Yet, they are still loaded for each export, and in our environment, we have more icon fonts as well, so it loads more items for every export.

Considering that, if I wanted to use a custom font, I should have that option enabled, but the engine would load all the other fonts that are not being used. Is it possible for a future update?

**Comment by martynasma** [comment]
> Considering that, if I wanted to use a custom font, I should have that option enabled, but the engine would load all the other fonts that are not being used. Is it possible for a future update?

Yeah, that would make sense.

I'm setting it up as a feature request. However, I can't provide an ETA, or even a promise it will be implemented :(

Will keep you posted, tho.

**Comment by martynasma** [comment]
OK, that was easy to implement. It will be in the next version:

```TypeScript
	/**
	 * A regular expression that will be matched against each URL of an external
	 * font being loaded. Font will only be loaded of regular expression matches.
	 *
	 * Has no effect of `useWebFonts` is set to `false`.
	 *
	 * ```TypeScript
	 * chart.exporting.webFontFilter = /pacifico|roboto/;
	 * ```
	 * ```JavaScript
	 * chart.exporting.webFontFilter = /pacifico|roboto/;
	 * ```
	 * ```JSON
	 * {
	 *   // ...
	 *   "exporting": {
	 *     // ...
	 *     "webFontFilter": /pacifico|roboto/
	 *   }
	 * }
	 * ```
	 *
	 * @since 4.10.17
	 */
	public webFontFilter: RegExp;
```

Will let you know when the new version is out.

**Comment by calvaradocl** [comment]
Hey @martynasma, that's great, we'll be looking forward to trying out this update.

**Comment by martynasma** [comment]
Fixed in [4.10.17](https://github.com/amcharts/amcharts4/releases/tag/4.10.17).

## [4.10.17] - 2021-03-19

### Added
- New setting for `Export`: `webFontFilter`. Allows it to set to a regular expression. If set only those external fonts that their URL match regular exporession will be loaded.
- New setting for `ForceDirectedSeries`: `showOnTick` (default `10`). Render series hidden until Xth tick.
- New setting for `XYCursor`: `snapOnPan` (default `true`). Indicates whether to "snap" zoom selection to equal periods when panning by cursor ends.

### Fixed
- Zooming of chart with multiple vertical axes of which one was inverted was incorrect.
- `minZoomCount` was not being taken into account when zooming with mouse wheel.
- JSON config: `exporting.dataFields` setting was being ignored.
- Creating chart before its container is ready, was resulting in extra element in DOM in some cases.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
